\PassOptionsToPackage{utf8}{inputenc}
\documentclass{bioinfo}
\copyrightyear{2018} \pubyear{2018}
\usepackage{natbib}
\usepackage[hidelinks]{hyperref}
\setcitestyle{round, author-year}

\access{Advance Access Publication Date: Day Month Year}
\appnotes{Original Paper}

\newcommand{\raxml}{RAxML}
\newcommand{\eggnog}{EggNOGG}
\newcommand{\epa}{EPA-ng}
\newcommand{\papara}{PaPaRa}
\newcommand{\phyma}{PhyloMagnet}

\begin{document}
\firstpage{1}

\subtitle{Sequence analysis}

\title[PhyloMagnet]{PhyloMagnet: Fast and accurate profiling of short-read meta-omics data using gene-centric phylogenetics}
\author[Sch\"on et al.]{Max E. Sch\"on\,$^{\text{\sfb 1,}*}$, Laura Eme\,$^{\text{\sfb 1}}$, Thijs J.G. Ettema\,$^{\text{\sfb 1,2,}*}$}
\address{$^{\text{\sf 1}}$Department of Cell and Molecular Biology, Science for Life Laboratory, Uppsala University, SE-75123 Uppsala, Sweden, $^{\text{\sf 2}}$Laboratory of Microbiology, Department of Agrotechnology and Food Sciences, Wageningen University, Stippeneng 4, 6708WE Wageningen, The Netherlands}


\corresp{$^\ast$To whom correspondence should be addressed.}

\history{Received on XXXXX; revised on XXXXX; accepted on XXXXX}

\editor{Associate Editor: XXXXXXX}

\abstract{\textbf{Motivation:} Metagenomic and metatranscriptomic sequencing analyses have become increasingly popular tools for  producing massive amounts of short-read data, often used for the reconstruction of draft genomes or the detection of (active) genes in microbial communities. Unfortunately, sequence assemblies of such datasets generally remain a computationally challenging task. Frequently, researchers are only interested in a specific group of organisms or genes; yet, the assembly of multiple datasets only to identify candidate sequences for a specific question is prohibitively slow, forcing researchers to select a subset of datasets to address their question (e.g. based on prior knowledge on the possible presence of the organism/gene). Here we present PhyloMagnet, an approach to screen meta-omics datasets for taxa and genes of interest using gene-centric assembly and phylogenetic placement of sequences.\\
\textbf{Results:} Using PhyloMagnet, we could identify up to 87\% of the genera in an in-vitro mock community with variable abundances, while the false positive predictions per single gene tree ranged from 0\% to 23\%. When applied to a group of metagenomes for which a set of MAGs have been published, we could detect the majority of the taxonomic labels that the MAGs had been annotated with. In a metatranscriptomic setting the phylogenetic placement of assembled contigs corresponds to that of transcripts obtained from transcriptome assembly.\\
\textbf{Availability:} PhyloMagnet is built using Nextflow, available at \href{https://github.com/maxemil/PhyloMagnet}{github.com/maxemil/PhyloMagnet} and is developed and tested on Linux. It is released under the open source GNU GPL license and documentation is available at \href{https://phylomagnet.readthedocs.io/en/latest/}{phylomagnet.readthedocs.io}.\\
\textbf{Contact:} \{max-emil.schon;thijs.ettema\}@icm.uu.se\\
\textbf{Supplementary information:} Supplementary data are available at \textit{Bioinformatics}
online.}

\maketitle

\section{Introduction}
High-throughput DNA sequencing has revolutionized biology, opening up new fields of research and enabling new fundamental insights in the life sciences. During the past decades, several sequencing technologies have been developed, each differing significantly in sequence read length, quality and throughput \citep{Mardis2017}. Applications comprise DNA shotgun sequencing as well as RNA sequencing of complex microbial communities, termed metagenomics and metatranscriptomics, respectively \citep{Mitchell2018}. 

Large environmental sequencing initiatives like the Tara Oceans project \citep{Sunagawa2015} have provided researchers with enormous amounts of metagenome data. Using recently developed genome-resolved or genome-centric metagenomic approaches, draft genomes or MAGs (Metagenome assembled genomes) of uncultured taxa can be assembled for the first time from shotgun metagenomic sequencing data of microbial communities \citep{Alneberg2014,Eren2015}. In order to apply those tools, however, metagenome assembly needs to be performed, which is computationally demanding and introduces additional challenges compared to single genome assembly such as the uneven coverage of contigs (contiguous sequences) from different organisms or the presence of micro-diversity \citep{Quince2017}. Together with the ever-growing sequencing capacity, it becomes increasingly taxing to identify which of the available datasets (publicly deposited or locally generated sequence datasets) actually contain sequence data of a given taxon or gene of interest.

Instead of assembling short reads into longer contigs, the taxonomic composition of a metagenomic or metatranscriptomic dataset can be assessed using microbiome profilers that classify reads directly. In general, these tools base their classification on the alignment of reads to reference sequences with a known taxonomy, and either work similar to the BLAST algorithm \citep[e.g. ][]{Huson2016,Truong2015} or use exact k-mer matches to such reference sequences to classify reads \citep[e.g. ][]{Ounit2015,Wood2014}. Development in this area is continuing in order to increase analysis speed while reducing memory footprint. Currently, DIAMOND is one of the fastest local aligners that has a sensitivity comparable to BLAST \citep{Buchfink2014}, and MetaCache is one of the fastest and most memory efficient k-mer based classifiers, using only a discriminatory subset of available k-mers \citep{Muller2017}. All of these approaches, however, are based on sequence similarity, which can be incongruent with the true phylogenetic relationship of sequences \citep{Smith2017}.

Traditional phylogenetic tools on the other hand offer several robust evolutionary models for both nucleic and amino acids that theoretically allow for a more reliable taxonomic assignment of sequences, but are slow compared to similarity-based methods, usually prohibiting their application to large metagenome datasets. In addition, short reads generally do not provide enough phylogenetic signal, leading to artifactual inferences \citep{Matsen2010}. Several tools have been developed to overcome these barriers by instead placing fragmentary sequences (particularly from amplicon sequencing data) onto a phylogenetic reference tree \citep{Matsen2010,Berger2011b,Barbera2019}.

Shotgun metagenomic or metatranscriptomic data is often analysed with a focus on gene rather than genome reconstruction, and is then usually called gene-centric. In this approach, the short reads or the assembled sequences are partitioned according to their affiliation to gene families. These methods can be used to determine which genes are present or actively transcribed in a sample, and can be combined with assemblers to reconstruct full-length sequences for a gene of interest. There exist several gene-centric targeted assemblers that perform de-novo reconstruction, e.g. via an overlap graph of candidate reads \citep{Kucuk2017,Pericard2017,Steinegger2018,Gruber-Vodicka2019}. While several of those only reconstruct the 16S gene or are limited to transcriptomic data, the MEGAN gene-centric assembler reconstructs contigs based on the alignment of reads to any reference protein sequence \citep{Huson2017}.

A recently published tool, GraftM, uses the ideas of phylogenetic placement and gene-centric metagenomics to taxonomically classify sequences of genes within metagenomes \citep{Boyd2018}. It is capable of placing either short-read sequences or pre-assembled metagenomic contigs onto a single reference tree at a time, but does not perform gene-centric assembly, which could increase phylogenetic signal of query sequences. Additionally, its reference trees can only be inferred using the extremely fast but often inaccurate maximum-likelihood-based tree inference program FastTree \citep{Price2010,Zhou2018}. Here, using an efficient workflow management system for parallel handling of both references and queries, gene-centric assembly, and robust phylogenetic inference, we show that we can outperform GraftM in terms of runtime and classification precision and sensitivity. 

The goals of the work presented here were to: 
\begin{enumerate}
\item[i] Create a computational workflow that could determine the presence of taxa of interest in large short-read datasets based on gene-centric assembly and robust phylogenetic inference, especially with the objective of selecting good candidate datasets for metagenomic assembly and genome-resolved metagenomics;
\item[ii] Create a workflow that uses state-of-the-art methods and is versatile and fast enough to accommodate a broad range of applications, while being modular in order to easily incorporate new approaches;
\item[iii] Compare the workflow’s performance in terms of computational footprint and sensitivity/precision to GraftM, another recently published tool with a similar application.
\end{enumerate}


\begin{methods}
\section{Implementation}
PhyloMagnet exploits the idea of gene-centric assembly \citep{Huson2017} to efficiently screen sequence datasets  of short reads for target genes, and to taxonomically classify them using phylogenetic placement. Below is a description of the analysis steps employed by the pipeline (see also Fig. \ref{fig:01}), which requires the following inputs: 
\begin{enumerate}

\item[a] One or several query short-read sequence files in FASTQ or FASTA format (potentially ‘raw’, untrimmed reads, see \ref{gc-assembly}), which correspond to the metagenomic or transcriptomic dataset to query;
\item[b] One or several homologous groups of reference proteins, each sequence annotated with its taxonomic affiliation (in the EggNOG format, containing ncbi’s taxonomy ID and a unique identifier, e.g. '70448.Q0P3H7').
\end{enumerate}

\subsection{Alignment and tree reconstruction of reference}
For each input group of reference sequences a multiple sequence alignment is computed using either MAFFT \citep{Katoh2013} or PRANK \citep{Loytynoja2010}, without applying any filtering or trimming methods. Then a reference tree is reconstructed using any of IQ-TREE \citep{Nguyen2015}, RAxML \citep{Stamatakis2014} or FastTree \citep{Price2010}, making it possible to choose the appropriate method for a specific analysis. This way the user can make a trade-off between speed and quality of the reference tree and choose the appropriate evolutionary model. Reference alignments and trees can be precomputed (e.g. on a local machine) and then provided to PhyloMagnet as a compressed reference package (e.g. on an HPC cluster). This also increases reproducibility, as such reference packages can be released alongside results.

\subsection{Alignment to reference protein sequences}
Identifiers from the EggNOG database, containing orthologous groups of protein sequences from all domains of life with functional annotations \citep{Huerta-cepas2016b}, can be specified to be used as input reference sequences. Alternatively, sets of homologous protein sequences curated by the user in FASTA format can be used. In order to check for the potential presence of homologs encoding these proteins of interest in the query metagenomes or metatranscriptomes, each of the short-read datasets given as input (see 2.a above) is then aligned to the collection of reference protein sequences using the DIAMOND aligner in blastX mode \citep{Buchfink2014}.

\begin{figure*}[!tpb]
\centerline{\includegraphics[width=.45\textwidth]{figures/Fig1.png}}
\caption{Illustration of the main steps in the PhyloMagnet workflow. (1) The required input is a dataset of short reads. (2) These reads are aligned against the complete set of protein references (blastX). (3) Using the protein alignments, homologous gene sequences are reconstructed for all groups of reference proteins. (4) The contigs are added to the reference protein alignments (5) and are subsequently placed onto the reference phylogenetic tree. (6) The results of the placement are summarized and the classification is visualized.}\label{fig:01}
\end{figure*}


\subsection{Gene-centric assembly of reads}\label{gc-assembly}
Using the gene-centric assembler implemented in MEGAN \citep{Huson2016,Huson2017}, reads are assembled into contigs. The assembly is performed independently for each orthologous group of reference proteins, and the available alignments of reads to the protein reference sequences of a group is used to infer overlaps between reads, thereby concatenating them into contigs. As only the aligned part (core) of each read is used for the assembly, no pre-processing such as adapter clipping or quality trimming is needed. The results are written to a FASTA file per orthologous group if any contig in that group passes the cut-off for the minimum length (200bp, can be adjusted if needed) that the gene-centric assembler uses. The assembled contigs are already in-frame and are subsequently translated into amino acid sequences using the standard genetic code.

\subsection{Phylogenetic placement of reconstructed protein sequences}
The assembled contigs are aligned to the alignments of each homologous reference group (maintaining the columns of the previously computed reference alignment), using the phylogeny-aware alignment tool PaPaRa \citep{Berger2011a}. This alignment of reference sequences and contigs is then used to place the contigs onto the reference tree using the evolutionary placement algorithm (EPA-ng) \citep{Berger2011b,Barbera2019}. In a final stage the tool gappa is used to annotate the internal branches of the reference tree and assign taxonomic labels to the contigs based on the likelihood weights of the placement \citep{Czech2018}. Then a summary list of taxonomic labels is created.

\subsection{‘Magnetizing’ trees and identifying candidate datasets}
The user can choose to specify taxonomic names (e.g. '\textit{Escherichia}') that should be used to filter (‘magnetize’) the list of all labels, specify a taxonomic rank (e.g. 'family'), or a combination of both. The occurrences of the chosen taxonomic labels are summarized per reference group and metagenomic or transcriptomic datasets  in order to assist manual decision of candidate datasets. Using the information of how many trees were positive for a taxon of interest as an approximation of coverage (see \ref{results}), the user could for example select datasets that display differential coverage for subsequent genome extraction, which often relies on such differences to group genome contigs together \citep{Albertsen2013,Alneberg2014}. 

\subsection{Availability}
PhyloMagnet is an open source software package and released under a GPLv3 license. It is written as a Nextflow \citep{DiTommaso2017} script and available on github (github.com/maxemil/PhyloMagnet). Several functions and utilities are implemented either in python or bash \citep{Dalke2009,Mckinney2010,Huerta-cepas2016a}. All needed dependencies are available as a singularity \citep{Kurtzer2017} container (singularity-hub.org/collections/978) and the documentation can be found on ReadTheDocs (phylomagnet.readthedocs.io).

\section{Benchmarking}
To evaluate the performance of our workflow and exemplify its potential uses, we performed three benchmark experiments using an in vitro mock community as well as environmental metagenomic and metatranscriptomic sequencing datasets. We chose the datasets such that we could compare the results produced by PhyloMagnet to reference genome mapping data \citep{Singer2016}, genomes extracted from metagenomes with taxonomic annotation \citep{Delmont2018} and an assembled metatranscriptome \citep{Frazier2017}, respectively. For details on command line parameters see the supplementary methods.

\subsection{Reference sequences}
To assess the general taxonomic composition of datasets we used a set of 16 ribosomal proteins (rp16) that are thought to represent reliable phylogenetic markers, as they should be vertically inherited throughout evolution and present in a single copy in most organisms \citep{Brown2015}. For this, we downloaded the corresponding sets of unaligned homologous sequences from the EggNOG database v4.5.1 \citep[][see Table \ref{X}]{Huerta-cepas2016b}.

As a second set of reference protein sequences, we used the set of 12 protein coding genes known to be present in chloroplast genomes of Dinophyceae \citep{Howe2008}. This phylum of single-celled algae can be found in a wide range of aquatic environment and notably contains coral symbionts within the genus \textit{Symbiodinium} \citep{Gomez2012}. For each of the genes we downloaded all available curated chloroplast encoded protein sequences for all phyla from UniProt \citep{Apweiler2004} as well as all available proteins from the Dinophyceae from the same database.

All reference groups were aligned using MAFFT E-INS-i \citep{Katoh2013} and reference trees were reconstructed using IQ-TREE \citep[under the LG+G+F model;][]{Nguyen2015}.

\subsection{Datasets}
The first dataset we selected was the MBARC-26 (Mock Bacteria ARchaea Community), an in vitro mock community of 23 bacterial and 3 archaeal strains (in 24 genera) with finished reference genomes that were pooled and sequenced on an Illumina HiSeq instrument \citep{Singer2016}. As the taxonomic classification is dependent on the reference sequences, we added orthologous sequences to the EggNOG rp16 references for those genera missing from the original EggNOG datasets. To avoid using identical sequences as references, we used available genomes from related species within the same genera to expand the rp16 references. The orthologous proteins were identified by performing HMMER \citep[v3.1b2;][]{Eddy2011} searches of the EggNOG rp16 reference alignments against the additional proteomes (see supplementary methods). These extended rp16 references were then used as references in PhyloMagnet in order to classify the MBARC-26 short-read data. 

As a second dataset we used several metagenomic datasets from the geographic region ‘Southern Ocean’ that are part of the several metagenomes from the Tara Oceans Initiative, as defined by \cite{Delmont2018}. We used the EggNOG rp16 references to assess taxonomic composition in those datasets and compared the results to the taxonomic classification of the MAGs reconstructed by \cite{Delmont2018}. The authors extracted 375 genome bins from these datasets, but only presented detailed information, including taxonomy, for 13 ‘non-redundant’ MAGs that passed several quality and completeness filters. To be able to compare our classification results to a more extensive set of extracted genome bins, we inferred taxonomic labels for those bins that were not part of the 13 non-redundant MAGs using the tool sourmash that uses k-mer matches to taxonomically classify genomes \citep{TitusBrown2016}.

Finally, we analysed the metatranscriptomes published by \citet{Frazier2017}, who sequenced mRNA from both healthy corals and such that are affected by so-called ‘bleaching’, a stress response in which \textit{Symbiodinium} symbionts are expulsed \citep{Howe2008}. We used the chloroplast protein references to search for Dinophyceae (and especially \textit{Symbiodinium}) sequences in the metatranscriptome data. We then compared the assembled sequences and their placement in the reference trees with the sequences from the metatranscriptomic assembly available at NCBI’s GEO database \citep{Barrett2012}. Similar to how the assembly was generated, we combined all of the 27 individual datasets by \citet{Frazier2017} into a single dataset for this analysis. In order to identify the transcripts in the assembly, we performed a tblastN search, querying the reference sequences against a database of the quality filtered transcripts. The identified sequences were then, analogous to how sequences are classified in PhyloMagnet, placed onto the reference tree with EPA-ng. 

\subsection{Comparison with GraftM}
We compared the performance of PhyloMagnet with that of the recently published tool GraftM \citep{Boyd2018}. GraftM also places sequences (either unassembled reads or pre-assembled contigs) onto a reference phylogeny using the tool pplacer, for which EPA-ng represents a scalable replacement that is able to handle larger amounts of data.
We created GraftM reference packages (gpkgs; containing the reference alignment, tree and the taxonomic annotation) from each of the extended rp16 references using the create command (see supplementary methods). We then used each gpkg to analyse the MBARC-26 dataset and recovered taxonomic classifications of the query sequences.
For both tools, we counted the number of genera that were correctly identified in each tree (true positives) as well as the number of genera that were identified even though they were not present in the MBARC-26 mock community (false positives). We also assessed the runtime and memory consumption of both tools for analysis of the full MBARC-26 dataset (50 Gb) as well as for subsamples of 1\% and 10\% (0.5 and 5 Gb, respectively).

\end{methods}

\section{Results}\label{results}
\subsection{MBARC-26}

We evaluated the performance of PhyloMagnet and GraftM to detect the presence of the 24 MBARC genera \citep[23 of those detectable, as Nocardiopsis was part of the pooled community but not present in the sequence data from][]{Singer2016} in the metagenomic dataset (Fig. \ref{fig:S1} and Table \ref{tab:S1}). The number of correctly detected as well as falsely reported genera are displayed in Fig. \ref{fig:02}. PhyloMagnet correctly identified up to 20 (87\%) of the MBARC genera and up to 7 (with an average of 2) false positive genera in all of the 16 trees. In contrast, GraftM identified a maximum of 9 (39\%) of the correct MBARC genera while giving up to 14 (with an average of 4) false positives for each tree. Some of the reported false positive and false negative errors of both PhyloMagnet and GraftM could be attributed to closely related and possibly unresolved taxonomic groups such as \textit{Escherichia}/\textit{Salmonella}, \textit{Thermobacillus}/\textit{Paenibacillus} or possibly \textit{Clostridium}/\textit{Ruminiclostridium}. Another confounding factor might be the well-known disagreement between phylogeny and taxonomy in some cases \citep[e.g. \textit{Escherichia}/\textit{Salmonella};][]{Retchless2010}.  Some taxa with very low abundance in the data (e.g. \textit{Corynebacterium} and \textit{Clostridium}) were picked up by GraftM but not PhyloMagnet, which is likely due to the fact that there are not enough reads to reconstruct longer contigs for theses taxa, impeding an identification by PhyloMagnet as we used the default cut-off implemented in the gene-centric assembler. In general we observe a correspondence between the percentage of mapped reads \citep{Singer2016} and the number of trees a genus was detected in (Fig. \ref{fig:S2}), suggesting that we can use the number of trees as a rough proxy for the abundance of a taxon in a dataset.  
When comparing results for the full dataset and the subsampled datasets, PhyloMagnet seems to profit immensely from the additional data, likely because the assembler can connect more reads and thus reconstruct more contigs above the length threshold.
In terms of runtime, PhyloMagnet is twice as fast as GraftM on 10 CPUs, making more efficient use of available computational resources. It uses, however, significantly more memory due to the requirements of MEGAN that performs the memory intensive sequence assembly, which GraftM does not include (see Fig. \ref{fig:S1}).

\begin{figure*}[!tpb]%figure1
\centerline{\includegraphics[width=.45\textwidth]{figures/Fig2.png}}
\caption{Classification results of PhyloMagnet and GraftM on the MBARC-26 dataset. True positive and false positive (on the negative y-axis including zero) values are shown for both PhyloMagnet (red) and GraftM (blue) and for each reference OG (x-axis). The 3 different dataset sizes are shown by lighter (1\%), middle (10\%) and darker (full dataset) shades of the respective color.} \label{fig:02}
\end{figure*}

\subsection{Tara Southern Ocean}
We could identify 65 taxa over the 8 datasets and 16 reference trees, whereof 21 were found in at least 4 trees for at least one dataset (Fig. \ref{fig:03}). These taxa cover all but one of the taxonomic groups for which genomic bins could be identified by \citet[][marked with an asterisk in Fig. \ref{fig:03}]{Delmont2018}. Most noticeable, the authors of that study recovered 6 non-redundant high-quality MAGs for the family Flavobacteriaceae which could be identified in every single tree for each dataset here as well. Further, one MAG was associated with each the Alteromonadaceae, the Rickettsiales and the Alphaproteobacteria as well as two Gammaproteobacteria. All of these were detected here except for the Rickettsiales, who could have been misidenfied as Pelagibacterales in the phylogenetic placement. Alternatively, the genomic bins could have been mislabelled as Rickettsiales and actually belong to the Pelagibacterales, as those two lineages commonly artifactually branch together in phylogenetic trees due to convergent genome streamlining resulting in a similar sequence composition bias \citep{Roger2017,Martijn2018,Rodriguez-Ezpeleta2012,Viklund2013}. It is very likely that the MAGs that were labelled as Gammaproteobacteria by \citet{Delmont2018} are actually members of the Piscirickettsiaceae or Porticoccaceae, which were both detected by PhyloMagnet in several individual datasets and the majority of single gene trees. Here we also recovered the additional taxonomic labels (Porticoccaceae, Rhodobacteraceae, Pelagibacteraceae, Cryomorphaceae) that could be assigned to raw genomic bins (which were not included in the original analyses as they did not pass quality and/or completeness thresholds) from the same study (Table \ref{tab:S2}). Some of the labels we recovered were not represented by any MAGs/Bins, indicating either false positive classification of contigs or a low abundance of the genomic DNA, such that no genome bins could be reconstructed by \citet{Delmont2018}.

\begin{figure*}[!tpb]%figure1
\centerline{\includegraphics[width=.45\textwidth]{figures/Fig3.png}}
\caption{Classification results of the Tara Southern Oceans datasets. The heatmap on the right shows the identified taxa for each of the eight datasets, and only taxa which could be identified in at least four trees in at least one dataset are shown. For each combination of dataset and taxon, the value represents the number of trees a sequence from that dataset has been labelled with this taxon name. On the left a taxonomic tree (as defined by the ncbi taxonomy database) showing the relationship of lineages is depicted. Lineages represented by non-redundant and raw bins from \citet{Delmont2018} are marked with an asterisk.}\label{fig:03}
\end{figure*}

\subsection{Coral bleaching}
Contigs classified as Symbiodiniaceae could be identified in 10 out of the 12 chloroplast gene reference trees. The number of contigs we reconstructed for each gene from the pooled sequencing data of 23 datasets ranged from 2 (\textit{psb}E) to as many as 169 (\textit{psb}A), whereas we could identify either one or two transcripts from the corresponding published transcriptome assembly for 9 out of the 12 genes. The two genes for which no contigs could be reconstructed were \textit{psb}I and \textit{pet}D, both missing in the assembled transcriptome as well, which is likely due to two distinct issues. The \textit{psb}I gene is only around 30 amino acid residues long, making contigs shorter than the default length cut-off of 200bp implemented in the gene-centric assembler. Besides, \textit{psb}I has never been identified, experimentally or computationally, in any \textit{Symbiodinium} species, but the identification within the Dinophyceae comes from the species \textit{Amphidinium operculatum} \citep{Nisbet2004,Barbrook2014}. As for \textit{pet}D, it seems that its transcription level is quite low, so that very few reads would have been sequenced, making assembly of contigs or transcripts virtually impossible \citep{Nisbet2008}. In those cases where transcripts could be identified, they were generally placed on the same branches or very close within the reference tree as were all of the corresponding contigs (Fig. \ref{fig:04}).

\begin{figure*}[!tpb]%figure1
\centerline{\includegraphics[width=.6\textwidth]{figures/Fig4.png}}
\caption{Phylogenetic placement of contigs and transcripts. As an example of the coral bleaching metatranscriptome analyses, the tree of the plastid gene \textit{psb}B is shown. The sequences reconstructed by PhyloMagnet (left) and the assembled and quality-filtered transcripts from \citet[][right]{Frazier2017} were placed onto the reference tree. All sequences were placed on branches sister to two \textit{Symbiodinium} sequences (U6EFR9 and A0A0A0N1X1).}\label{fig:04}
\end{figure*}

\section{Conclusion}
We have shown that by applying phylogenetic placement methods to protein sequences that were reconstructed from short-read sequencing data, we can accurately identify short-read sequence datasets that contain sequences for genes or taxa of interest. We compared our tools to a similar approach that does not use gene-centric assembly and demonstrated that our approach is faster and has a higher precision and sensitivity (at the price of  consuming more memory). 

Several lineages could be identified from single gene trees that match the results of a genome-resolved metagenomic study, showcasing how our tool can be used to screen the contents of a metagenomic dataset before applying metagenome assembly and binning methods. We are convinced that this will enable more researchers to use genome-resolved metagenomics in their research, such as exploring the diversity of genomes of a specific clade or to genomes that contain a specific metabolic pathway, as the more computationally demanding tasks could be limited to those candidate datasets that were identified by PhyloMagnet.

The example application of our gene-centric phylogenetic approach to metatranscriptomic data shows how it can be of general purpose also for this type of data. Similar to metagenomic data, it can be useful in detecting taxa of interest, but also in assessing the phylogenetic diversity and overall transcription and activity of genes of interest.


%\section*{Acknowledgements}


\section*{Funding}
MES is funded by the Horizon 2020 research and innovation program under the Marie Sk\l{}odowska-Curie ITN project SINGEK (http://www.singek.eu/; grant agreement No. H2020-MSCA-ITN-2015-675752). LE was funded by the European Union’s Horizon 2020 research and innovation programme under the Marie Sk\l{}odowska-Curie grant agreement No 704263.

%\bibliographystyle{natbib}
\bibliographystyle{bioinformatics}
\bibliography{PM.bib}

\newpage
\renewcommand\thefigure{S\arabic{figure}}
\setcounter{figure}{0}    
\begin{figure*}[!h]%figure1
\centerline{}
\caption{Classification results of PhyloMagnet and GraftM on the full MBARC-26 dataset. Values and colors correspond to the number of rp16 trees a genus was identified in. Those genera that are part of the MBARC-26 community are written in bold.}\label{fig:S1}
\end{figure*}

\begin{figure*}[!h]%figure1
\centerline{}
\caption{Computational footprint of PhyloMagnet and GraftM for the complete MBARC-26 dataset (50\,Gb), the subsample of 1\% (5\,Gb) and of 10\% (0.5\,Gb). top: runtime on 10 CPUs. bottom: peak memory usage.}\label{fig:S2}
\end{figure*}

\renewcommand\thetable{S\arabic{table}}
\setcounter{table}{0}    

\begin{table}[!t]
\caption{Organisms included in the MBARC-26 dataset. For each organism, current taxonomy (species, genus and family as given in the ncbi taxonomy), ncbi assembly ID and percentage mapped illumina reads as presented in \citet{Singer2016} are shown.}\label{tab:S1} 
\end{table}

\begin{table}[!t]
\caption{Taxonomic annotation of MAGs extracted from the Tara southern oceans dataset by \citet{Delmont2018} The inferered taxonomic labels from domain to species level (as far as available) are provided for the 11 prokaryotic nonredundant MAGs (inferred with GraftM) as well as for those additional raw bins where a taxonomic annotation could be inferred with sourmash.}\label{tab:S2} 
\end{table}
\end{document}
